src/cli/cli_udp.cpp

```diff
@@ -256,6 +256,8 @@ template <> otError UdpExample::Process<Cmd("send")>(Arg aArgs[])
     otMessageInfo     messageInfo;
     otMessageSettings messageSettings = {mLinkSecurityEnabled, OT_MESSAGE_PRIORITY_NORMAL};
 
+    VerifyOrExit(otUdpIsOpen(GetInstancePtr(), &mSocket), error = OT_ERROR_INVALID_STATE);
+
```

Comment:
This looks good. Thanks.

One suggestion: How about we perform this check in the `Udp::SendTo()` method instead? This would ensure the check is applied consistently regardless of how the API is used.

Reply:
Yes, that would definitely make sense and I was also thinking about that for a few days. But in the end I decided to leave `Udp::SendTo()` as it is. My main reason for that was to conserve resources on constrained devices by avoiding searching the linked list of sockets which is done by `otUdpIsOpen()` method. On the other hand, I thought developers who are already familiar with the API use a check before calling `Udp::SendTo()` if the nature of their application demands it.

But if you think my fears about performance and resources are exagerrated, I of course support to move this check to `Udp::SendTo()` method since that could improve stability of applications.

Reply:
> My main reason for that was to conserve resources on constrained devices by avoiding searching the linked list of sockets which is done by otUdpIsOpen() method. On the other hand, I thought developers who are already familiar with the API use a check before calling Udp::SendTo() if the nature of their application demands it.

Thanks. Yes. I agree that it'd be better to avoid linked-list search on every `SendTo()` call.  Adding this check in `cli` command sound resonable.